src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAGIC_NUMBER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        int randomNumber = random.nextInt(5);
+
+        switch (randomNumber) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomSquare();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomRectangle();
+            case 4:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                break;
+        }
+        return getRandomFigure();
```

Comment:
```suggestion
        switch (randomNumber) {
            case 0:
                return getRandomCircle();
            case 1:
                return getRandomSquare();
            case 2:
                return getRandomRightTriangle();
            case 3:
                return getRandomRectangle();
            default:
                return getRandomIsoscelesTrapezoid()
                break;
        }
```

and you won't need private getRandomFigure method